src/cmd/health.js

```diff
@@ -37,94 +47,315 @@ function print_usage() {
+    const error_code = await this.get_error_code(service_status, pid, rsyslog.service_status, response_code);
+    const bucket_details = await this.get_bucket_storage_status(this.config_root);
+    const account_details = await this.get_account_storage_status(this.config_root);
+    const health = {
+      service_name: NSFS_SERVICE,
       status: service_health,
       memory: memory,
+      error: error_code,
       checks: {
-          service: {
-              service_status: service_status,
-              pid: pid,
+          services: [{
+            name: NSFS_SERVICE,
+            service_status: service_status,
+            pid: pid,
           },
+          {
+            name: RSYSLOG_SERVICE,
+            service_status: rsyslog.service_status,
+            pid: rsyslog.pid,
+          }],
           endpoint: {
-              endpoint_response: status_code,
+            endpoint_state
           },
+          invalid_accounts: account_details.invalid_storages,
+          valid_accounts: account_details.valid_storages,
+          invalid_buckets: bucket_details.invalid_storages,
+          valid_buckets: bucket_details.valid_storages,
       }
+    };
+    return health;
+  }
+
+  async get_error_code(nsfs_status, pid, rsyslog_status, endpoint_response_code) {
+    if (nsfs_status !== "active" || pid === "0") {
+      return health_errors.NSFS_SERVICE_FAILED;
+    } else if (rsyslog_status !== "active") {
+      return health_errors.RSYSLOG_SERVICE_FAILED;
+    } else if (endpoint_response_code === 500) {
+      return health_errors.NSFS_ENDPOINT_FAILED;
+    } else if (endpoint_response_code === 404) {
+      return health_errors.NSFS_ENDPOINT_FORK_MISSING;
+    }
+  }
+
+  async get_service_state(service_name) {
+    const service_status = await os_util.exec('systemctl show -p ActiveState --value ' + service_name, {
+      ignore_rc: true,
+      return_stdout: true,
+      trim_stdout: true,
+    });
+    const pid = await os_util.exec('systemctl show --property MainPID --value ' + service_name, {
+      ignore_rc: true,
+      return_stdout: true,
+      trim_stdout: true,
+    });
+    return { service_status: service_status, pid: pid };
+  }
+
+  async make_endpoint_health_request(url_path) {
+    const response = await make_https_request(
+      { HOSTNAME,
+        port: this.https_port,
+        path: url_path,
+        method: 'GET',
+        rejectUnauthorized: false,
+      });
+    if (response && response.statusCode === 200) {
+      const buffer = await read_stream_join(response);
+      const body = buffer.toString('utf8');
+      return JSON.parse(body);
+    }
+  }
+
+  async get_endpoint_response() {
+    let url_path = '/total_fork_count';
+    const worker_ids = [];
+    let total_fork_count = 0;
+    let response_code;
+      try {
+        const fork_count_response = await this.make_endpoint_health_request(url_path);
+        if (!fork_count_response) {
+          return {
+            response_code: 500,
```

Comment:
aren't we supposed to have here  `response: fork_response_code[response_code]` ?